### ISSUE TITLE
Leanplum iPhone A/B test issue

### ISSUE THREAD

**Issue by nbhasin2** [mozilla-mobile/firefox-ios#6656]
Scenario 1: Don't get the update callback

a) User downloads Firefox Beta from testflight
b) Expectation: Leanplum to send A/B test variable to present onboarding screen
c) Result: I never get the A/B test variable and always see default screen

Scenario 2: Get the update callback

a) User downloads Firefox Beta from testflight
b) On first launch as soon as app launches I move it to background
c) Then when I bring the app back to foreground I see proper onboarding screen
d) Expectation: Leanplum to send A/B test variable to present proper onboarding screen
e) Result: I get the A/B test variable and see proper variant of the screen only after sending app to background and then bringing it to foreground

**Comment by garvankeeley** [comment]
What about IPA builds installed from BB?

**Comment by nbhasin2** [comment]
IPA downloaded directly from BB works 100% of the time
iPAD Testflight also works 100% of the time

**Comment by nbhasin2** [comment]
**Verified in v26.0 (18099)**

Leanplum has few responses when the variable gets changed. I was using onVariablesChanged before. However it took longer than expected and at times we didn't receive a response for when the app was downloaded through testflight. 

Following the doc - https://docs.leanplum.com/reference#callbacks
All A/B test variables should get synced right after **Leanplum.start** callback. Taking that in consideration all code on the v26.x branch was refactored and a callback inside start was added that when it finished we receive it on the Browser View Controller (BVC).

Along with the callback we also added a 2sec delay to accommodate for situations like 
- No internet connection 
- Leanplum start already sent callback but BVC was too late to initialize thus we check the variables in BVC after 2sec as well. 

LP Callback order set
```
        Leanplum.onStartResponse{ (success:Bool) in
            //3rd
        }
        lpShowOnboarding?.onValueChanged({
            //1st
        })
        Leanplum.onVariablesChanged {
            //2nd
        }
        Leanplum.start {
            //4th
        }
```
